### PR TITLE
fix: Discounts quantity matching fix

### LINKED ISSUE #5865: Able to book more discounted or access restricted tickets than allowed
Even if the organizer only allows 1 Discount Ticket. The discount ticket code still works after 1 ticket was sold. The same applies for access codes.

Expected: Discount Tickets and Access Codes should only work for the number of tickets specified.

E.g.:
* if the user chooses to buy two tickets, only one ticket should be discounted, if only one discount ticket is left.
* if all discount tickets are already sold out, the discount option should not show up on the ticket page. Instead there should be a message "discount tickets sold out" when a discount ticket code is entered in the UI.

Screenshot from 2019-05-03 16-51-37

### PR DESCRIPTION
**Pull request by mrsaicharan1**
Fixes #5865 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

- [x] All the functions created/modified in this PR contain relevant docstrings.

#### Short description of what this resolves:
- Matches the discount quantity correctly

#### Changes proposed in this pull request:

-  Removed complex query of fetching discount code IDs and splitting for ids
- Matches the discount quantity as expected

### CODE REVIEW

**Comment by mrsaicharan1** [comment]
@uds5501 The previous query had some discrepancies where there was a string being split for ticket ids. I've simplified the query as for now and also, I've tried my best as to why the quantity match was going wrong here but I'm not sure what's wrong. Can you please check this?

**Comment by mrsaicharan1** [comment]
@uds5501 The previous query had some discrepancies where there was a string being split for ticket ids. I've simplified the query as for now and also, I've tried my best as to why the quantity match was going wrong here but I'm not sure what's wrong. Can you please check this? Also, do you know what `tickets_number` means?

File: app/api/helpers/ticketing.py [hunk 1]
```diff
@@ -24,14 +25,14 @@ def get_order_expiry():
...
+            old_holders += order.tickets_count
         for holder in ticket_holders:
-            ticket_holder = TicketHolder.query.filter_by(id=holder).one()
-            if ticket_holder.ticket.id in discount_code.tickets.split(","):
-                qty += 1
-        if (qty + old_holders) <= discount_code.tickets_number and \
-            discount_code.min_quantity <= qty <= discount_code.max_quantity:
+            qty += 1
+        if discount_code.min_quantity <= (qty + old_holders) <= discount_code.max_quantity:
+            print(qty + old_holders)
```
Comment (uds5501) [inline app/api/helpers/ticketing.py]: @mrsaicharan1 any progress on it?

**Comment by mrsaicharan1** [comment]
@uds5501 You can take this up.

**Pull request closed without merging**